pytest_ansible/module_dispatcher/v2.py

```diff
@@ -0,0 +1,146 @@
+import warnings
+import ansible.constants
+import ansible.utils
+import ansible.errors
+
+from pkg_resources import parse_version
+from ansible.plugins.callback import CallbackBase
+from ansible.executor.task_queue_manager import TaskQueueManager
+from ansible.playbook.play import Play
+from ansible.cli import CLI
+from pytest_ansible.logger import get_logger
+from pytest_ansible.module_dispatcher import BaseModuleDispatcher
+from pytest_ansible.results import AdHocResult
+from pytest_ansible.errors import AnsibleConnectionFailure
+
+has_ansible_v2 = parse_version(ansible.__version__) >= parse_version('2.0.0')
+
+if not has_ansible_v2:
+    raise ImportError("Only supported with ansible-2.* and newer")
+
+log = get_logger(__name__)
+
+
+class ResultAccumulator(CallbackBase):
+
+    """FIXME."""
+
+    def __init__(self, *args, **kwargs):
+        super(ResultAccumulator, self).__init__(*args, **kwargs)
+        self.contacted = {}
+        self.unreachable = {}
+
+    def v2_runner_on_failed(self, result, *args, **kwargs):
+        self.contacted[result._host.get_name()] = result._result
+
+    v2_runner_on_ok = v2_runner_on_failed
+
+    def v2_runner_on_unreachable(self, result):
+        self.unreachable[result._host.get_name()] = result._result
+
+    @property
+    def results(self):
+        return dict(contacted=self.contacted, unreachable=self.unreachable)
+
+
+class ModuleDispatcherV2(BaseModuleDispatcher):
+
+    """Pass."""
+
+    required_kwargs = ('inventory', 'inventory_manager', 'variable_manager', 'host_pattern', 'loader')
+
+    def has_module(self, name):
+        return ansible.plugins.module_loader.has_plugin(name)
+
+    def _run(self, *module_args, **complex_args):
+        """The API provided by ansible is not intended as a public API."""
```

Comment:
In fact, this code does live in ansible.  However, `pytest_ansible` abstracts the `_run` method to allow customizing support for different versions of Ansible.